src/tests/compat_api.rs

```diff
@@ -504,3 +505,50 @@ mod bootstrap {
         expect_event!(event_rx0, Event::BootstrapFailed);
     }
 }
+
+mod when_no_message_received_within_inactivity_period {
+    use super::*;
+
+    #[test]
+    fn when_heartbeats_disabled_tcp_peer_emits_lost_peer_event() {
+        let config = unwrap!(ConfigFile::new_temporary());
+        unwrap!(config.write()).listen_addresses = vec![tcp_addr!("0.0.0.0:0")];
+        let (mut service1, event_rx1) = service_with_config(config);
+        service1.disable_peer_heartbeats();
+
+        unwrap!(service1.start_listening());
+        let service1_addr = expect_event!(event_rx1, Event::ListenerStarted(addr) => addr);
+        unwrap!(service1.set_accept_bootstrap(true));
+
+        let (event_tx2, event_rx2) = event_sender();
+        let config2 = unwrap!(ConfigFile::new_temporary());
+        unwrap!(config2.write()).bootstrap_cache_name = Some(util::bootstrap_cache_tmp_file());
+        unwrap!(config2.write()).hard_coded_contacts =
+            vec![PeerInfo::new(service1_addr, service1.public_id())];
+        let sk2 = SecretId::new();
+        let mut service2 = unwrap!(compat::Service::with_config(
+            event_tx2,
+            config2,
+            sk2.clone()
+        ));
+        service2.disable_peer_heartbeats();
+
+        unwrap!(service2.start_bootstrap(HashSet::new(), CrustUser::Client));
+        expect_event!(event_rx2, Event::BootstrapConnect(_peer_id, _));
+        expect_event!(event_rx1, Event::BootstrapAccept(_peer_id, _));
+
+        thread::sleep(Duration::from_millis(5 * INACTIVITY_TIMEOUT_MS));
```

Comment:
Can we not just wait until we receive lost peer from the channel rather than compulsorily waiting for this duration ? Seems a little wasteful otherwise to just block the test for this duration - think the timeout is pretty low for tests but we shouldn't rely on it unless we are testing the event is not fired before the timeout or something but probably that's not the case here.
We can ofc use [rx_timeout(inactivity_timeout + 10ms or something)](https://doc.rust-lang.org/std/sync/mpsc/struct.Receiver.html#method.recv_timeout) if we are concerned about hanging in case of a buggy production code.